Flaky test: AutoAdvanceTest > onCardChange reloads settings

```
AutoAdvanceTest > onCardChange reloads settings FAILED
    java.lang.AssertionError: Verification failed: call 1 of 1: ActionListener(listener#63).onAutoAdvanceAction(eq(SHOW_REMINDER), any())) was not called
        at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:65)
        at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:39)
        at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:50)
        at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:63)
        at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:32)
        at io.mockk.MockKDsl.internalCoVerify(API.kt:152)
        at io.mockk.MockKKt.coVerify(MockK.kt:267)
        at io.mockk.MockKKt.coVerify$default(MockK.kt:257)
        at com.ichi2.anki.ui.windows.reviewer.autoadvance.AutoAdvanceTest$onCardChange reloads settings$1.invokeSuspend(AutoAdvanceTest.kt:186)
        at com.ichi2.anki.ui.windows.reviewer.autoadvance.AutoAdvanceTest$onCardChange reloads settings$1.invoke(AutoAdvanceTest.kt)
        at com.ichi2.anki.ui.windows.reviewer.autoadvance.AutoAdvanceTest$onCardChange reloads settings$1.invoke(AutoAdvanceTest.kt)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$1.invokeSuspend(TestBuilders.kt:317)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
        at kotlinx.coroutines.test.TestDispatcher.processEvent$kotlinx_coroutines_test(TestDispatcher.kt:24)
        at kotlinx.coroutines.test.TestCoroutineScheduler.tryRunNextTaskUnless$kotlinx_coroutines_test(TestCoroutineScheduler.kt:99)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$workRunner$1.invokeSuspend(TestBuilders.kt:326)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:263)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:94)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:70)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:10)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:309)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0$default(TestBuilders.kt:304)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0$default(Unknown Source)
        at com.ichi2.anki.ui.windows.reviewer.autoadvance.AutoAdvanceTest.onCardChange reloads settings(AutoAdvanceTest.kt:164)
```

https://github.com/ankidroid/Anki-Android/actions/runs/24698737502/job/72237187830

--- a/AnkiDroid/src/test/java/com/ichi2/anki/ui/windows/reviewer/autoadvance/AutoAdvanceTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/ui/windows/reviewer/autoadvance/AutoAdvanceTest.kt
@@ -169,13 +169,14 @@ class AutoAdvanceTest {
                     durationToShowQuestionFor = 2.seconds,
                 )
 
-            coEvery { AutoAdvanceSettings.createInstance(any()) } returns initialSettings andThen newSettings
+            coEvery { AutoAdvanceSettings.createInstance(1L) } returns initialSettings
 
             autoAdvance = createAutoAdvance()
             autoAdvance.isEnabled = true
             testScheduler.advanceUntilIdle()
 
             val newCard = mockk<Card>(relaxed = true)
+            coEvery { AutoAdvanceSettings.createInstance(2L) } returns newSettings
             every { newCard.currentDeckId() } returns 2L
             autoAdvance.onCardChange(newCard)
             testScheduler.advanceUntilIdle()

